sky/tools/release_engine.py

```diff
@@ -36,42 +37,61 @@ def git_revision(cwd):
     ], cwd=cwd).strip()
 
 
-class Artifact(object):
-    def __init__(self, category, name):
-        self.category = category
-        self.name = name
+GS_ZIP_URL = 'gs://mojo/sky/shell/%(commit_hash)s/artifacts.zip'
+GS_FILE_URL = 'gs://mojo/sky/shell/%(commit_hash)s/%(config)s/%(name)s'
 
 
-GS_URL = 'gs://mojo/sky/%(category)s/%(config)s/%(commit_hash)s/%(name)s'
+# Paths of the artifacts that will be packaged into a zip file.
+ZIP_ARTIFACTS = {
+    'android-arm': [
+        'chromium-debug.keystore',
+        'icudtl.dat',
+        'dist/shell/SkyShell.apk',
+        'dist/shell/flutter.mojo',
+        'gen/sky/shell/shell/classes.dex',
+        'gen/sky/shell/shell/shell/libs/armeabi-v7a/libsky_shell.so',
+    ],
+    'linux-x64': [
+        'dist/shell/icudtl.dat',
+        'dist/shell/sky_shell',
+        'dist/shell/sky_snapshot',
+        'dist/shell/flutter.mojo',
+    ],
+}
 
 
-ARTIFACTS = {
+# Paths of the artifacts that will be uploaded to GCS as individual files.
+FILE_ARTIFACTS = {
     'android-arm': [
-        Artifact('shell', 'SkyShell.apk'),
-        Artifact('shell', 'flutter.mojo'),
-        Artifact('shell', 'libflutter_library.so'),
+        'dist/shell/flutter.mojo',
+        'dist/shell/libflutter_library.so',
     ],
     'linux-x64': [
-        Artifact('shell', 'icudtl.dat'),
-        Artifact('shell', 'sky_shell'),
-        Artifact('shell', 'sky_snapshot'),
-        Artifact('shell', 'flutter.mojo'),
-        Artifact('shell', 'libflutter_library.so'),
-    ]
+        'dist/shell/flutter.mojo',
+        'dist/shell/libflutter_library.so',
+    ],
 }
 
 
+def find_missing_artifacts(config, config_root):
+    result = []
+    for artifact_map in [ZIP_ARTIFACTS, FILE_ARTIFACTS]:
+        for artifact_path in artifact_map[config]:
+            full_path = os.path.join(config_root, artifact_path)
+            if not os.path.exists(full_path):
+                result.append(full_path)
+    return result
+
+
 def upload_artifacts(dist_root, config, commit_hash):
-    for artifact in ARTIFACTS[config]:
-        src = os.path.join(artifact.category, artifact.name)
-        dst = GS_URL % {
-            'category': artifact.category,
+    for artifact_path in FILE_ARTIFACTS[config]:
+        dst = GS_FILE_URL % {
             'config': config,
             'commit_hash': commit_hash,
-            'name': artifact.name,
+            'name': os.path.basename(artifact_path),
         }
-        z = ','.join([ 'mojo', 'dat' ])
-        run(dist_root, ['gsutil', 'cp', '-z', z, src, dst])
+        z = ','.join([ 'mojo', 'dat', 'so', 'dex' ])
```

Comment:
We should probably adjust this list to match what we're actually uploading here.